webapp/src/app/admin/contacts/contacts.component.ts

```diff
@@ -430,16 +485,20 @@ export interface IUserCred {
     view: string;
     groups: string[];
     groupsAvail: string[];
+    recovery: string;
 }
 
 @Component({
     selector: "app-user-cred",
     templateUrl: "user-cred.html",
 })
 export class UserCredComponent {
+
+    static noRecovery = "No recovery";
```

Comment:
No need for a special string value, we can use `undefined`.

```diff
diff --git a/webapp/src/app/admin/contacts/contacts.component.ts b/webapp/src/app/admin/contacts/contacts.component.ts
index 8d83980..d88c167 100644
--- a/webapp/src/app/admin/contacts/contacts.component.ts
+++ b/webapp/src/app/admin/contacts/contacts.component.ts
@@ -72,7 +72,7 @@ export class ContactsComponent implements OnInit {
     async createContact(view?: string, groups?: string[]) {
         const groupsInstAvail = await this.uData.contact.getGroups();
         const groupsAvail = groupsInstAvail.map((g) => g.darc.description.toString());
-        const creds: IUserCred = {alias: "", email: "", view, groups, groupsAvail, recovery: null};
+        const creds: IUserCred = {alias: "", email: "", view, groups, groupsAvail};
         // Search if the proposed groups are really available to the user
         if (groups && groups.find((group) => groupsAvail.find((g) => g === group) === undefined)) {
             return Promise.reject("unknown group");
@@ -155,7 +155,7 @@ export class ContactsComponent implements OnInit {
                             progress(60, "Updating User Information");
                             await this.uData.bc.sendTransactionAndWait(ctx);
 
-                            if (result.recovery && result.recovery !== UserCredComponent.noRecovery) {
+                            if (result.recovery !== undefined) {
                                 progress(70, "Adding recovery");
                                 Log.print("adding recovery");
                                 await nu.contact.updateOrConnect();
@@ -485,7 +485,7 @@ export interface IUserCred {
     view: string;
     groups: string[];
     groupsAvail: string[];
-    recovery: string;
+    recovery?: string;
 }
 
 @Component({
@@ -494,7 +494,6 @@ export interface IUserCred {
 })
 export class UserCredComponent {
 
-    static noRecovery = "No recovery";
     views = Data.views;
     showGroups: boolean;
     showViews: boolean;
@@ -509,8 +508,8 @@ export class UserCredComponent {
             data.view = Data.views[0];
         }
         data.groups = [];
-        this.recoveryGroups = [UserCredComponent.noRecovery].concat(data.groupsAvail);
-        data.recovery = this.recoveryGroups[0];
+        this.recoveryGroups = data.groupsAvail;
+        data.recovery = undefined;
     }
 
     cancel(): void {
```

Reply:
I want the user to be able to chose a recovery-group and then say "oh dang, in fact I don't want recovery", and remove the recovery group. So there needs to be a special entry for "no entry". At least that's how it's in the material demo:

https://material.angular.io/components/select/overview